### PR TITLE
Convert copy to r-value

### PR DESCRIPTION
**Pull request by iagarwa**
Avoid bigger objects copy

### CODE REVIEW

**Comment by iagarwa** [comment]
@MalaySanghi @mahesh-attarde please review

**Review by RKSimon — COMMENTED** [review]
Did you find these cases by manual inspection or did you use a static analyzer?

File: llvm/lib/MC/DXContainerPSVInfo.cpp [hunk 1]
```diff
@@ -168,7 +168,7 @@ void PSVRuntimeInfo::finalize(Triple::EnvironmentType Stage) {
   DXConStrTabBuilder.add(EntryName);
 
   DXConStrTabBuilder.finalize();
-  for (auto ElAndName : zip(SignatureElements, SemanticNames)) {
+  for (auto &&ElAndName : zip(SignatureElements, SemanticNames)) {
```
Comment (llvm-beanz) [inline llvm/lib/MC/DXContainerPSVInfo.cpp:171]: The copy here would just be copying a `std::tuple` of iterators (which are pointers). That doesn’t seem like something that should cause any performance issues, and this doesn’t match other idiomatic use cases of `llvm::zip`.

**Comment by iagarwa** [comment]
These copy are only copying the iterations which are smaller in size, so need of the change. closing the PR

**Comment by iagarwa** [comment]
These copy are only copying the iterations which are smaller in size, so need of the change. closing the PR

**Pull request closed without merging**